lib/datadog/core/configuration/option.rb

```diff
@@ -9,17 +9,36 @@ class Option
         attr_reader \
           :definition
 
+        # Option setting precedence. Higher number means higher precedence.
+        module Precedence
+          REMOTE_CONFIGURATION = 2
+          PROGRAMMATIC = 1
```

Comment:
Minor: I wonder if there should exist an internal local override mode that is the highest priority (even if we don't document it)? Even if only for testing, it seems weird that the highest priority is remote configuration and that's the latest word on the subject.

Reply:
That can make sense!
I don't think there's a use case for that yet, because we reset settings in between each test run.

But I'm not opposed to adding it if a use case arises.

Reply:
Yeah, good point, no use adding more complexity for a speculative use case.